src/indicators/on_balance_volume.rs

```diff
@@ -0,0 +1,162 @@
+use std::fmt;
+
+use crate::errors::*;
+use crate::{Close, Next, Reset, Volume};
+
+/// On Balance Volume (OBV).
+///
+/// The OBV is an volume and price based oscillator which gives cumulative total volumes.
+/// OBV measures buying and selling pressure as a cumulative indicator,
+/// adding volume on up days and subtracting it on down days.
+///
+/// # Formula
+///
+/// If the closing price is above the prior close price then:
+/// Current OBV = Previous OBV + Current Volume
+///
+/// If the closing price is below the prior close price then:
+/// Current OBV = Previous OBV  -  Current Volume
+///
+/// If the closing prices equals the prior close price then:
+/// Current OBV = Previous OBV
+///
+/// Where:
+///
+/// obv - on the balance volume
+///
+/// # Example
+///
+/// ```
+/// use ta::indicators::OnBalanceVolume;
+/// use ta::{Next, DataItem};
+///
+/// let mut obv = OnBalanceVolume::new().unwrap();
+/// let di = DataItem::builder()
+///             .high(3.0)
+///             .low(1.0)
+///             .close(2.0)
+///             .open(1.5)
+///             .volume(1000.0)
+///             .build().unwrap();
+/// obv.next(&di);
+///
+/// ```
+/// # Links
+/// * [On Balance Volume, Wikipedia] (https://en.wikipedia.org/wiki/On-balance_volume)
+/// * [On Balance Volume, stockcharts] (https://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:on_balance_volume_obv)
+
+#[derive(Debug, Clone)]
+pub struct OnBalanceVolume {
+    obv: f64,
+    prev_close: f64,
+}
+
+impl OnBalanceVolume {
+    pub fn new() -> Result<Self> {
```

Comment:
Since this indicator does not accept any parameters, there is not possibility for `Err` result.
I would prefer return the indicator directly instead of `Result<Self>`.

Reply:
Changed in new commit